torch/_inductor/utils.py

```diff
@@ -1190,7 +1190,8 @@ def use_cpp_packed_gemm_template(layout, mat1, mat2):
         and micro_gemm is not None
         and mat1.get_stride()[-1] == 1  # TODO(jgong5): support transposed input
         and isinstance(mat2, ir.StorageBox)
-        and mat2.is_module_buffer()
+        and (mat2.is_module_buffer() or not require_constant_mat2)
+        and mat1.get_name() != mat2.get_name()
```

Comment:
Can you explain the problem when mat1 and mat2 point to the same buffer?

Reply:
Problem was that the list of input nodes was only one element, but the accompanying indices were 2+ elements. Has been changed to just repeat the single input node.